agent/api/task/service_connect_validation.go

```diff
@@ -0,0 +1,293 @@
+		if err := validateCIDR(trimmedIpv4cidr, "IPv4 CIDR"); err != nil {
+			return err
+		}
+	}
+
+	// validate IPV6CIDR if it exists
+	if scEgressConfig.VIP.IPV6CIDR != "" {
+		trimmedIpv6cidr := strings.TrimSpace(scEgressConfig.VIP.IPV6CIDR)
+		if err := validateCIDR(trimmedIpv6cidr, "IPv6 CIDR"); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// validateDnsConfig validates the service connnect DNS config.
+func validateDnsConfig(scDnsConfligList []DNSConfigEntry, scEgressConfig *EgressConfig) error {
+	if len(scDnsConfligList) == 0 && scEgressConfig != nil {
+		// DNS config is a field which associates to egress config
+		return fmt.Errorf("no service connect DNS config. The DNS config is required when the egress config is not empty")
+	}
+
+	for _, dnsEntry := range scDnsConfligList {
+		// HostName is a required field
+		if dnsEntry.HostName == "" {
+			return fmt.Errorf("no hostname in the DNS config entry hostname=%s, address=%s", dnsEntry.HostName, dnsEntry.Address)
+		}
+
+		// Address is a required field
+		if dnsEntry.Address == "" {
+			return fmt.Errorf("no address in the DNS config entry hostname=%s, address=%s", dnsEntry.HostName, dnsEntry.Address)
+		}
+
+		// validate the address is a valid IPv4/IPv6 address
+		if err := validateAddress(dnsEntry.Address); err != nil {
+			return fmt.Errorf("invalid address in the DNS config entry hostname=%s, address=%s: %w", dnsEntry.HostName, dnsEntry.Address, err)
+		}
+	}
+
+	return nil
+}
+
+// validateAddress validates the passed address is a valid IPv4/IPv6.
+func validateAddress(address string) error {
+	if ip := net.ParseIP(address); ip == nil {
+		return fmt.Errorf("address=%s is not a valid textual representation of an IP address", address)
+	}
+	return nil
+}
+
+// validateCIDR validates the passed CIDR is a valid IPv4/IPv6 CIDR.
+func validateCIDR(cidr, version string) error {
+	if _, _, err := net.ParseCIDR(cidr); err != nil {
+		return fmt.Errorf("cidr=%s is not a valid %s: %w", cidr, version, err)
+	}
+	return nil
+}
+
+// validateIngressConfig validates the service connect ingress config based on network mode.
+func validateIngressConfig(scIngressConfigList []IngressConfigEntry, taskNetworkMode string) error {
+	// ingress config can be empty since an ECS service can only act as a client
+	if len(scIngressConfigList) == 0 {
+		return nil
+	}
+
+	// AWSVPC network mode
+	switch taskNetworkMode {
+	case BridgeNetworkMode:
+		if err := validateBridgeIngressConfig(scIngressConfigList); err != nil {
+			return err
+		}
+	case AWSVPCNetworkMode:
+		if err := validateAwsVpcIngressConfig(scIngressConfigList); err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("service connect does not support for %s newtork mode", taskNetworkMode)
+	}
+
+	return nil
+}
+
+// validateAwsVpcIngressConfig validates the service connect ingress config for awsvpc network mode.
+func validateAwsVpcIngressConfig(scIngressConfigList []IngressConfigEntry) error {
```

Comment:
The only difference between this and Bridge mode validation is around the presence of hostPort. Could we combine these into one so that we dont duplicate all of the other validation?

Reply:
Refactor ingress config validation in the commit 3d8d95406e74fc25d3c00d384a11cd5cd4ae55f9. Please take a look on the agent/api/task/service_connect_validator.go. Thanks